TradeItIosTicketSDK2/TradeItAlertManager.swift

```diff
@@ -0,0 +1,122 @@
+    var linkBrokerUIFlow = TradeItLinkBrokerUIFlow(linkedBrokerManager: TradeItLauncher.linkedBrokerManager)
+    let tradeItErrorManager = TradeItErrorManager()
+
+    /*
+     func showTradeItErrorResultAlert(onViewController viewController: UIViewController,
+                                                       errorResult: TradeItErrorResult,
+                                                       onAlertDismissed: () -> Void)
+     */
+    func showGenericError(tradeItErrorResult tradeItErrorResult: TradeItErrorResult,
+                            onViewController viewController: UIViewController,
+                                             onFinished: () -> Void = {}) {
+        var alertTitle = ""
+        var alertMessage = ""
+        var alertActionTitle = ""
+
+        if let shortMessage = tradeItErrorResult.shortMessage {
+            alertTitle = shortMessage
+        }
+
+        if let longMessages = tradeItErrorResult.longMessages {
+            alertMessage = (longMessages as! [String]).joinWithSeparator(" ")
+        }
+
+        alertActionTitle = "OK"
+
+        self.showOn(viewController: viewController,
+                    withAlertTitle: alertTitle,
+                    withAlertMessage: alertMessage,
+                    withAlertActionTitle: alertActionTitle,
+                    onAlertActionTapped: onFinished)
+    }
+
+    func show(tradeItErrorResult tradeItErrorResult: TradeItErrorResult,
+                onViewController viewController: UIViewController,
+                withLinkedBroker linkedBroker: TradeItLinkedBroker,
+                                 onFinished : () -> Void) {
+        var alertTitle = ""
+        var alertMessage = ""
+        var alertActionTitle = ""
+        var onAlertActionTapped: () -> Void
+        let onAlertActionRelinkAccount: () -> Void = {
+            self.linkBrokerUIFlow.presentRelinkBrokerFlow(
+                inViewController: viewController,
+                linkedBroker: linkedBroker,
+                onLinked: { (presentedNavController: UINavigationController) -> Void in
+                    presentedNavController.dismissViewControllerAnimated(true, completion: nil)
+                    linkedBroker.refreshAccountBalances(
+                        onFinished: onFinished
+                    )
+                },
+                onFlowAborted: { (presentedNavController: UINavigationController) -> Void in
+                    onFinished()
+                }
+            )
+        }
+
+        if tradeItErrorManager.isBrokerAuthenticationError(tradeItErrorResult) {
+            alertTitle = "Update Login"
+            alertMessage = "There seem to be a problem connecting with your \(linkedBroker.linkedLogin.broker) account. Please update your login information."
+            alertActionTitle = "Update"
+            onAlertActionTapped = onAlertActionRelinkAccount
+            self.showOn(viewController: viewController,
```

Comment:
What is `showOn`? What are we showing on the view controller? Should this be `showErrorOn` or `showAlertOn`?